rspec-mode.el

```diff
@@ -646,14 +667,24 @@ file if it exists, or sensible defaults otherwise."
 
 (defun rspec--shell-quote-local (file)
   (let ((remote (file-remote-p file))
+        (docker (rspec-docker-p))
         (vagrant (rspec-vagrant-p)))
     (shell-quote-argument
      (cond
       (remote (substring file (length remote)))
+      (docker (replace-regexp-in-string (regexp-quote (rspec-project-root))
+                                         rspec-docker-cwd file))
       (vagrant (replace-regexp-in-string (regexp-quote (rspec-project-root))
                                          rspec-vagrant-cwd file))
       (t  file)))))
 
+(defun rspec--docker-wrapper (command)
+  (if (rspec-docker-p)
+      (format "docker exec %s bash -c \"%s\""
```

Comment:
Problem with `docker-compose run` is, that it does not work properly under Windows (only with `-d` option, which might be sufficient).
Maybe a variable `rspec-docker-command` which holds the command to use could be defined. This variable could have a default value of "docker-compose run" and the user could change it to "docker exec" by needs. This even would allow the use of other commands if necessary.